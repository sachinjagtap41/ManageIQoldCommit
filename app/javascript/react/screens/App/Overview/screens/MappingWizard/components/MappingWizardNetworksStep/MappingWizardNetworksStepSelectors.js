export const groupByUidEms = (networks = []) =>
  networks.reduce(
    (networksMap, network) => ({
      ...networksMap,
      [network.uid_ems]: networksMap[network.uid_ems]
        ? [...networksMap[network.uid_ems], network]
        : [network]
    }),
    {}
  );
