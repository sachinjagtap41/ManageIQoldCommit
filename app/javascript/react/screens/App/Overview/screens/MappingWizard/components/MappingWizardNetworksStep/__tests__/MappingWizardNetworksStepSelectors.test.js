import { sourceClusterNetworks } from '../mappingWizardNetworksStep.fixtures';
import { groupByUidEms } from '../MappingWizardNetworksStepSelectors';

test('groupByUidEms creates a mapping of networks organized by uid_ems', () => {
  const sourceNetworks = sourceClusterNetworks[0].lans;

  expect(groupByUidEms(sourceNetworks)).toMatchSnapshot();
});
