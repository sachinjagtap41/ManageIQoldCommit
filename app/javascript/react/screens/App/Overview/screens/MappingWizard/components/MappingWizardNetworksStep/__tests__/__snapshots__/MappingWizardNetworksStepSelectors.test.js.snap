// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`groupByUidEms creates a mapping of networks organized by uid_ems 1`] = `
Object {
  "VMWareCluster1-vm_network1": Array [
    Object {
      "allow_promiscuous": null,
      "computed_allow_promiscuous": null,
      "computed_forged_transmits": null,
      "computed_mac_changes": null,
      "created_on": "2017-06-01T04:24:18Z",
      "forged_transmits": null,
      "href": "http://localhost:3000/api/lans/10000000000007",
      "id": "1-1",
      "mac_changes": null,
      "name": "VMWareCluster1-vm_network1",
      "parent_id": null,
      "switch_id": "1",
      "tag": "23",
      "uid_ems": "VMWareCluster1-vm_network1",
      "updated_on": "2017-06-01T04:24:18Z",
    },
    Object {
      "allow_promiscuous": null,
      "computed_allow_promiscuous": null,
      "computed_forged_transmits": null,
      "computed_mac_changes": null,
      "created_on": "2017-06-01T04:24:18Z",
      "forged_transmits": null,
      "href": "http://localhost:3000/api/lans/10000000000008",
      "id": "1-2",
      "mac_changes": null,
      "name": "VMWareCluster1-vm_network1",
      "parent_id": null,
      "switch_id": "2",
      "tag": "24",
      "uid_ems": "VMWareCluster1-vm_network1",
      "updated_on": "2017-06-01T04:24:18Z",
    },
  ],
  "VMWareCluster1-vm_network2": Array [
    Object {
      "allow_promiscuous": null,
      "computed_allow_promiscuous": null,
      "computed_forged_transmits": null,
      "computed_mac_changes": null,
      "created_on": "2017-06-01T04:24:18Z",
      "forged_transmits": null,
      "href": "http://localhost:3000/api/lans/10000000000009",
      "id": "1-3",
      "mac_changes": null,
      "name": "VMWareCluster1-vm_network2",
      "parent_id": null,
      "switch_id": "1",
      "tag": "24",
      "uid_ems": "VMWareCluster1-vm_network2",
      "updated_on": "2017-06-01T04:24:18Z",
    },
  ],
  "VMWareCluster1-vm_network3": Array [
    Object {
      "allow_promiscuous": null,
      "computed_allow_promiscuous": null,
      "computed_forged_transmits": null,
      "computed_mac_changes": null,
      "created_on": "2017-06-01T04:24:18Z",
      "forged_transmits": null,
      "href": "http://localhost:3000/api/lans/10000000000010",
      "id": "1-4",
      "mac_changes": null,
      "name": "VMWareCluster1-vm_network3",
      "parent_id": null,
      "switch_id": "1",
      "tag": "24",
      "uid_ems": "VMWareCluster1-vm_network3",
      "updated_on": "2017-06-01T04:24:18Z",
    },
  ],
}
`;
